keras_nlp/models/preprocessor.py

```diff
@@ -128,17 +136,23 @@ def from_preset(
                 "choose a particular task class, e.g. "
                 "`keras_nlp.models.BertPreprocessor.from_preset()`."
             )
-        config_file = "tokenizer.json"
-        preset_cls = check_config_class(preset, config_file=config_file)
-        if preset_cls is not cls.tokenizer_cls:
+
+        # Ensure loading is not from an incorrect class.
+        tokenizer_preset_cls = check_config_class(
+            preset, config_file=TOKENIZER_CONFIG_FILE
+        )
+        if tokenizer_preset_cls is not cls.tokenizer_cls:
```

Comment:
this whole if block can got in the second if/else below. we don't need to check any of this if we are loading a `preprocessor.json` directly

Reply:
Isn't this needed to prevent loading from an incorrect class? Something like this:
```
keras_nlp.models.GPT2Preprocessor.from_preset("bert_tiny_en_uncased")
```

Reply:
Discussed offline!
Done!